src/axom/quest/io/STLWriter.hpp

```diff
@@ -0,0 +1,116 @@
+// Copyright (c) 2017-2025, Lawrence Livermore National Security, LLC and
+// other Axom Project Developers. See the top-level LICENSE file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+
+#ifndef QUEST_STLWRITER_HPP_
+#define QUEST_STLWRITER_HPP_
+
+// Axom includes
+#include "axom/config.hpp"
+#include "axom/core/Macros.hpp"
+#include "axom/mint/mesh/Mesh.hpp"
+
+#include <string>
+
+namespace axom
+{
+namespace quest
+{
+/*!
+ * \class STLWriter
+ *
+ * \brief A simple STL writer for the faces of Mint meshes.
+ *
+ * STL (STereoLithography) is a common file format for triangle meshes.
+ * It encodes a "soup of triangles" by explicitly listing the coordinate
+ * positions of the three vertices of each triangle.
+ *
+ * \note Any faces that contain more than 3 vertices will be treated
+ *       like a triangle fan and will result in multiple triangles.
+ *       No care is taken to prevent triangle overlaps for non-convex faces.
+ */
+class STLWriter
+{
+public:
+  /*!
+   * \brief Constructor.
+   */
+  STLWriter() = default;
+
+  /*!
+   * \brief Constructor.
+   *
+   * \param filename The name of the file to write.
+   * \param binary Whether or not to write a binary STL file.
+   */
+  STLWriter(const std::string& filename, bool binary = false);
+
+  /*!
+   * \brief Destructor.
+   */
+  ~STLWriter() = default;
+
+  /*!
+   * \brief Sets the name of the file to write.
+   * \param [in] fileName the name of the file to write.
+   */
+  void setFileName(const std::string& fileName) { m_fileName = fileName; }
+
+  /*!
+   * \brief Sets whether to use binary output.
+   * \param [in] binary True to write binary output; false to write ASCII.
+   */
+  void setBinary(bool binary) { m_binary = binary; }
+
+  /*!
+   * \brief Writes the mesh into an STL file.
+   * \param [in] mesh pointer to the mesh to write.
+   * \pre path to input file has been set by calling `setFileName()`
+   * \return status set to zero on success; set to a non-zero value otherwise.
+   */
+  int write(const mint::Mesh* mesh);
+
+// The following members are protected (unless using CUDA)
+#if !defined(__CUDACC__)
+protected:
+#endif
+
+  /*!
+   * \brief Compute the number of triangles produced for the input mesh.
+   *
+   * \param mesh The input mesh.
+   *
+   * \return The number of triangles.
+   */
+  IndexType getNumberOfTriangles(const mint::Mesh* mesh) const;
+
+  /*!
+   * \brief Determines whether the input mesh looks topologically 2D since
+   *        the mesh dimension seems to reflect the number of coordinate components.
+   *
+   * \param mesh The mesh to test.
+   *
+   * \return True if the mesh seems 2D, false otherwise.
+   */
+  bool isTopologically2D(const mint::Mesh* mesh) const;
```

Comment:
Should the `mesh` be a class member? I'm assuming you did it this way for consistency w/ the STLReader.

Reply:
I put `mesh` as a class member. Someday we can include a conduit::Node* member too.